packages/aws-cdk/lib/api/aws-auth/sdk.ts

```diff
@@ -51,22 +68,22 @@ export interface ISDK {
    */
   removeCustomUserAgent(userAgentData: string): void;
 
-  lambda(): AWS.Lambda;
-  cloudFormation(): AWS.CloudFormation;
-  ec2(): AWS.EC2;
-  iam(): AWS.IAM;
-  ssm(): AWS.SSM;
-  s3(): AWS.S3;
-  route53(): AWS.Route53;
-  ecr(): AWS.ECR;
-  ecs(): AWS.ECS;
-  elbv2(): AWS.ELBv2;
-  secretsManager(): AWS.SecretsManager;
-  kms(): AWS.KMS;
-  stepFunctions(): AWS.StepFunctions;
-  codeBuild(): AWS.CodeBuild;
-  cloudWatchLogs(): AWS.CloudWatchLogs;
-  appsync(): AWS.AppSync;
+  lambda(): Lambda;
+  cloudFormation(): CloudFormation;
+  ec2(): EC2;
+  iam(): IAM;
+  ssm(): SSM;
+  s3(): S3;
+  route53(): Route53;
+  ecr(): ECR;
+  ecs(): ECS;
+  elbv2(): ElasticLoadBalancingV2;
+  secretsManager(): SecretsManager;
+  kms(): KMS;
+  stepFunctions(): SFN;
+  codeBuild(): CodeBuild;
+  cloudWatchLogs(): CloudWatchLogs;
+  appsync(): AppSync;
```

Comment:
Let's not change any of these methods for now. Create new ones, that return the corresponding service client (e.g., `S3Client`) and incrementally switch to these new methods. Once no one is using the old methods anymore, we delete them.

Reply:
Donezo

Reply:
actually I realized I took a completely different approach where I just created a completely updated `SDKv3` class with all of these updated and then kept the old `SDK` class. My thought was that when migrating everything else we'd swap it all to SDKv3 when migrating then when everything is done using `SDK` we delete it and rename `SDKv3` to `SDK`. 

Let me know if you think this is a bad approach and you'd rather see me keep the same class and simply swap out just one service at a time.

Reply:
Yes, a brand new class and interface also works. My only suggestion now is to include in this PR a change in some other file that uses the `ISDK` interface, just to smoke test the change. There's a lot of fiddling about with error handling and whatnot here. Let's make sure they still work in v3.